include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -130,6 +131,22 @@ __find_start_point(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_
     return _split_point_t<_Index>{*__res, __index_sum - *__res + 1};
 }
 
+template <typename _Rng1DataType, typename _Rng2DataType, typename = void>
+struct __can_use_ternary_op : std::false_type
+{
+};
+
+template <typename _Rng1DataType, typename _Rng2DataType>
+struct __can_use_ternary_op<_Rng1DataType, _Rng2DataType,
+                            std::void_t<decltype(true ? std::declval<_Rng1DataType>()
+                                                      : std::declval<_Rng2DataType>())>>
+    : std::true_type
+{
+};
+
+template <typename _Rng1, typename _Rng2>
+constexpr static bool __can_use_ternary_op_v = __can_use_ternary_op<_Rng1, _Rng2>::value;
```

Comment:
nitpick:  
Naming of `_Rng[1/2]`.  Honestly I wouldn't mind all these just being `_T1`, `_T2`.  They don't need to care about how they are called.